x/concentrated-liquidity/swaps_tick_cross_test.go

```diff
@@ -1238,3 +1238,154 @@ func (s *SwapTickCrossTestSuite) TestSwapOutGivenIn_PriceLimit_TickCross_ZeroFor
+
+		// 3TS position is in range.
+		s.assertPositionInRange(poolId, positionThreeTSLowerTick, positionThreeTS.upperTick)
+
+		// Liquidity from amounts for position 3TS is computed using the in-range option.
+		validateInRangeLiquidityFromAmounts(currentSqrtPrice, positionThreeTSLowerSqrtPrice, positionThreeTSUpperSqrtPrice)
+
+		// 2TS position is out of range.
+		s.assertPositionOutOfRange(poolId, positionTwoTSLowerTick, positionTwoTS.upperTick)
+
+		// 2TS position should consist of token zero only as it is to the right of the active range.
+		liquidity02TS := math.Liquidity0(DefaultAmt0, currentSqrtPrice, positionTwoTSUpperSqrtPrice)
+		actualLiquidity2Ts := math.GetLiquidityFromAmounts(pool.GetCurrentSqrtPrice(), positionTwoTSLowerSqrtPrice, s.tickToSqrtPrice(positionTwoTS.upperTick), DefaultAmt0, DefaultAmt1)
+		s.Require().Equal(liquidity02TS, actualLiquidity2Ts)
+
+		// Reset suite context
+		s.Ctx = setupCtx
+	})
+
+	s.Run("one for zero", func() {
+		// Set temporary cache context on the suite
+		// so that we can reuse test helpers without mutating
+		// the setup
+		cacheCtx, _ := s.Ctx.CacheContext()
+		s.Ctx = cacheCtx
+
+		/// Compute tokenIn amount necessary to reach the upper tick of the 3TS position.
+		const (
+			isZeroForOne                  = false
+			shouldStayWithinTheSameBucket = false
+		)
+		tokenIn, _, _ := s.computeSwapAmounts(poolId, sdk.Dec{}, positionTwoTSUpperTick, isZeroForOne, shouldStayWithinTheSameBucket)
+
+		s.swapOneForZeroRight(poolId, sdk.NewCoin(tokenOne, tokenIn.TruncateInt()))
+
+		// Assert that the desired tick is reached.
+		s.assertPoolTickEquals(poolId, positionTwoTSUpperTick)
+
+		// Refetch pool
+		pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+		s.Require().NoError(err)
+
+		currentSqrtPrice := pool.GetCurrentSqrtPrice()
+
+		// Current sqrt price equals to 2TS position's upper tick sqrt price.
+		s.Require().True(currentSqrtPrice.Equal(positionTwoTSUpperSqrtPrice))
+
+		// Current sqrt price is below the upper tick sqrt price of the 3TS position.
+		s.Require().True(currentSqrtPrice.LT(positionThreeTSUpperSqrtPrice))
+
+		// 3 TS position is in range.
+		s.assertPositionInRange(poolId, positionThreeTSLowerTick, positionThreeTS.upperTick)
+
+		// Liquidity from amounts for 3TS position is computed using the in-range option.
+		validateInRangeLiquidityFromAmounts(currentSqrtPrice, positionThreeTSLowerSqrtPrice, positionThreeTSUpperSqrtPrice)
+
+		// 2TS position is out of range.
+		s.assertPositionOutOfRange(poolId, positionTwoTSLowerTick, positionTwoTS.upperTick)
+
+		// 2TS should consist of token one only as it is to the right of the active range.
+		liquidity1Pos3 := math.Liquidity1(DefaultAmt1, currentSqrtPrice, positionTwoTSLowerSqrtPrice)
+		actualLiquidityPos3 := math.GetLiquidityFromAmounts(pool.GetCurrentSqrtPrice(), positionTwoTSLowerSqrtPrice, positionTwoTSUpperSqrtPrice, DefaultAmt0, DefaultAmt1)
```

Comment:
Might be worth mentioning in a comment that this is a somewhat loose check since `GetLiquidityFromAmounts` can return `Liquidity1` in the in-range case if it's the smaller liquidity amount

Reply:
In that case, I think the values would be different still. Note this:

```go
# in range option
liquidity1 := Liquidity1(amount1, sqrtPrice, sqrtPriceA)

# out-of-range option
liquidity = Liquidity1(amount1, sqrtPriceB, sqrtPriceA)
```

The in-range option goes from current sqrt price to lower sqrt price.

The out-of-range option goes from lower to upper sqrt prices